docs/configuration.md

```diff
@@ -0,0 +1,219 @@
+# Configuration
+
+## Manifest Templates
+
+* [Manifest with Logs, APM, process, and metrics collection enabled.][1]
+* [Manifest with Logs, APM, and metrics collection enabled.][2]
+* [Manifest with Logs and metrics collection enabled.][3]
+* [Manifest with APM and metrics collection enabled.][4]
+* [Manifest with Cluster Agent.][5]
+
+## All configuration options
+
+The following table lists the configurable parameters for the `DatadogAgent`
+resource. For example, if you wanted to set a value for
+`agent.credentials.apiKey` and `agent.credentials.appKey`, your `DatadogAgent`
```

Comment:
nit: we should probably recommend using secrets in the example instead of passing the creds in plain text